pkg/infra/remotecache/redis_storage.go

```diff
@@ -22,9 +22,14 @@ func parseRedisConnStr(connStr string) (*redis.Options, error) {
 	keyValueCSV := strings.Split(connStr, ",")
 	options := &redis.Options{Network: "tcp"}
 	for _, rawKeyValue := range keyValueCSV {
-		keyValueTuple := strings.Split(rawKeyValue, "=")
+		keyValueTuple := strings.SplitN(rawKeyValue, "=", 2)
 		if len(keyValueTuple) != 2 {
-			return nil, fmt.Errorf("incorrect redis connection string format detected for '%v', format is key=value,key=value", rawKeyValue)
+			if strings.HasPrefix(rawKeyValue, "password") {
+				// don't log the password
+				return nil, fmt.Errorf("incorrect redis connection string format detected for 'password', format is key=value,key=value")
+			} else {
```

Comment:
As you may have noticed lint fails here with the following error:

> pkg/infra/remotecache/redis_storage.go
>   (30, 11)  https://revive.run/r#indent-error-flow  if block ends with a return statement, so drop this else and outdent its block

you have to remote the `else` and just return the error as proposed.